pkg/webhook/cluster_build_template.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2017 Google Inc. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"errors"
+
+	"github.com/mattbaird/jsonpatch"
+
+	"github.com/knative/build/pkg/apis/build/v1alpha1"
+	"github.com/knative/build/pkg/logging"
+)
+
+func (ac *AdmissionController) validateClusterBuildTemplate(ctx context.Context, _ *[]jsonpatch.JsonPatchOperation, old, new genericCRD) error {
+	_, tmpl, err := unmarshalClusterBuildTemplates(ctx, old, new)
+	if err != nil {
+		return err
+	}
+
+	if err := validateSteps(tmpl.Spec.Steps); err != nil {
```

Comment:
Can we share these between the two types of build templates?

Reply:
added a template_common.go file with the common functions called by both types of templates